### DISCUSSION TITLE
Boundary condition for cylinder

### DISCUSSION THREAD

**Discussion by AyushBoral** [xcompact3d/Incompact3d#217] [Q&A] [answered]
What is the variable phi and phi1 in BC_Cylinder.f90? Is it the boundary condition at the cylinder wall? If not, then which variable to edit to include the effect of rotation in the immersed boundary?

**Comment by slaizet** [comment, marked as answer]
Hi,
These arrays are for a scalar field. To include the effect of rotation you need to change the velocity field values.
Best,
Sylvain

**Reply by AyushBoral** [reply]
Thanky you for the help!
So I need to change the values of ux, uy and uz at the boundaries and the appropriate initial conditions?

**Reply by slaizet** [reply]
something like that yes!

**Reply by AyushBoral** [reply]
I tried to implement the boundary conditions on the velocity by the following modification to the BC-Cylinder.f90 file. I found that the boundary condition was not correctly implemented, and I was getting unusually high values of divergence of velocity.
![Screenshot 2023-08-01 234515](https://github.com/xcompact3d/Incompact3d/assets/140534517/5720e8c5-385a-4dad-976a-ed25ac82dd10)
![Screenshot 2023-08-01 234433](https://github.com/xcompact3d/Incompact3d/assets/140534517/5385d7f4-33d5-49c0-af85-40956bbba652)
